src/napari_plugin_manager/qt_package_installer.py

```diff
@@ -62,6 +63,28 @@ def _constraints_file(cls) -> str:
         return f.name
 
 
+class NapariUvInstallerTool(UvInstallerTool):
+    @staticmethod
+    def constraints() -> list[str]:
+        """
+        Version constraints to limit unwanted changes in installation.
+        """
+        return [f'napari=={_napari_version}']
```

Comment:
Hm. So the only risk for now is if the environment has Numba installed and the plugin needs a NumPy newer than the installed one. But it is a problem as well for pip.